src/gui/guiFormSpecMenu.cpp

```diff
@@ -4174,12 +4215,25 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 			s_count = list_s->getItem(s.i).count;
 		} while(0);
 
+		// True if the hovered slot is the selected slot
 		bool identical = m_selected_item && s.isValid() &&
 			(inv_selected == inv_s) &&
 			(m_selected_item->listname == s.listname) &&
 			(m_selected_item->i == s.i);
 
-		ButtonEventType button = BET_LEFT;
+		// True if the hovered slot is empty
+		bool empty = s.isValid() && list_s->getItem(s.i).empty();
+
+		// True if the hovered item would stack with the selected item
+		bool matching = false;
+		if (m_selected_item && s.isValid()) {
+			ItemStack a = list_selected->getItem(m_selected_item->i);
+			ItemStack b = list_s->getItem(s.i);
+			if (a.name == b.name && a.wear == a.wear && a.metadata == a.metadata)
```

Comment:
`a.itemFits(b, nullptr, itemdef)` should be more reliable than manually checking properties.

Reply:
That function returns false if `b` doesn't completely fit in `a`, so it can't be used. Also the `==` operator compares the count of the items, so that can't be used either.

Reply:
I implemented `ItemStack::stacksWith()`, I think that name makes more sense for the use case: `a.stacksWith(b)`.